grails-app/controllers/org/pih/warehouse/api/LocationApiController.groovy

```diff
@@ -241,7 +241,15 @@ class LocationApiController extends BaseDomainApiController {
         }
         def parentLocation = existingLocation.parentLocation
         if (parentLocation) {
+            existingLocation.parentLocation = null
             parentLocation.removeFromLocations(existingLocation)
+            parentLocation.save(flush: true)
+        }
+        def zone = existingLocation.zone
+        if (zone) {
+            existingLocation.zone = null
+            zone.removeFromLocations(existingLocation)
+            zone.save(flush: true)
```

Comment:
This shouldn't be necessary since there's no association between zone and internal locations so this is basically a no-op. I assume this is in response to an issue you had (something like "deleted object would be re-saved"). In the future, can you add any issues you encounter to the ticket a) so there's an audit trail of the issue and b) so we can discuss it.